tree/treeplayer/src/TBranchProxy.cxx

```diff
@@ -18,13 +18,31 @@ of the autoloading of branches as well as all the generic setup routine.
 #include "TLeaf.h"
 #include "TBranchElement.h"
 #include "TBranchObject.h"
+#include "TCollection.h" // TRangeStaticCast
 #include "TStreamerElement.h"
 #include "TStreamerInfo.h"
+#include "ROOT/InternalTreeUtils.hxx" // GetFileNamesFromTree, GetTreeFullPaths
+
+#include <string>
+#include <string_view>
 
 ClassImp(ROOT::Detail::TBranchProxy);
 
 using namespace ROOT::Internal;
 
+namespace {
+/**
+ * \brief Find if the input branch name is the prefix for other sub-branches in the same tree.
+ */
+bool AreThereSubBranches(std::string_view parentBrName, TTree &tree)
+{
+   for (auto *leaf : ROOT::Detail::TRangeStaticCast<TLeaf>(tree.GetListOfLeaves()))
+      if (std::string(leaf->GetName()).rfind(std::string(parentBrName) + ".", 0) != std::string::npos)
+         return true;
+   return false;
```

Comment:
Consider creating the string only once:
```suggestion
   const std::string branchPrefix{ std::string(parentBrName) + "." };
   for (auto *leaf : ROOT::Detail::TRangeStaticCast<TLeaf>(tree.GetListOfLeaves()))
      if (std::string(leaf->GetName()).rfind(branchPrefix, 0) != std::string::npos)
         return true;
   return false;
```
or even 
```suggestion
   const std::string branchPrefix{ std::string(parentBrName) + "." };
   for (auto *leaf : ROOT::Detail::TRangeStaticCast<TLeaf>(tree.GetListOfLeaves()))
      if (branchPrefix.compare(0, branchPrefix.length(), leaf->GetName())
         return true;
   return false;
```

Reply:
The `compare` solution does not work as is:
```
root [0] std::string prefix{"trkl.beg."};
root [4] prefix.compare(0, prefix.size(), "trkl.begrndm")
(int) -68
```

I still need both elements of the comparison to be strings
```
root [0] std::string prefix{"trkl.beg."};
root [1] std::string brname{"trkl.beg.subbranch"};
root [11] brname.compare(0, prefix.size(), prefix)
(int) 0
```

Reply:
But I agree with creating the prefix string only once

Reply:
For compare, I meant:
```
if (branchPrefix.compare(0, branchPrefix.length(), leaf->GetName()) == 0)
```

Reply:
I see, then I believe `-1` will be the correct return value because of the trailing dot
```
root [0] std::string branchPrefix{"trkl.beg."};
root [1] std::string brName{"trkl.beg.x"};
root [2] branchPrefix.compare(0, branchPrefix.size(), brName.c_str())
(int) -1
```

Reply:
I don't see a good description of the numerical value (besides negative, zero, positive) in the standard.  So maybe (using the name in your example) we should use:
```
branchPrefix.compare(0, branchPrefix.size(), brName.c_str(), branchPrefix.size())
```
which returns 0 when we have a match.